tests/Unit/TolerantVariableReferenceFinderTest.php

```diff
@@ -0,0 +1,201 @@
+            '<?php <sr>$v<>ar1 = 5; $func = function() use (<sr>$var1) { };',
+            $uri,
+        ];
+
+        yield 'scope: anonymous function: inside' => [
+            '<?php <sr>$v<>ar1 = 5; $func = function() use (<sr>$var1) { $var2 = <sr>$var1; };',
+            $uri,
+        ];
+
+        yield 'scope: anonymous function: inside selection' => [
+            '<?php <sr>$var1 = 5; $func = function() use (<sr>$var1) { $var2 = <sr>$v<>ar1; };',
+            $uri,
+        ];
+
+        yield 'scope: anonymous function: only inside' => [
+            '<?php $var1 = 2; $func = function() { <sr>$v<>ar1 = 5; $var2 = <sr>$var1 + 10; };',
+            $uri,
+        ];
+
+        yield 'scope: anonymous function: only outside' => [
+            '<?php <sr>$va<>r1 = 2; $func = function() { $var1 = 5; $var2 = $var1 + 10; }; $var2 = <sr>$var1 / 4;',
+            $uri,
+        ];
+
+        yield 'scope: inside class method' => [
+            '<?php class C1 { function M1(<sr>$var1) { <sr>$v<>ar1 = 5; $var2 = <sr>$var1 + 10; } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: select argument' => [
+            '<?php class C1 { function M1(<sr>$va<>r1) { <sr>$var1 = 5; $var2 = <sr>$var1 + 10; } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function + use, click inside' => [
+            '<?php class C1 { function M1() { <sr>$var1 = 10; $f = function() use (<sr>$var1) { <sr>$v<>ar1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function + use, click outside' => [
+            '<?php class C1 { function M1() { <sr>$v<>ar1 = 10; $f = function() use (<sr>$var1) { <sr>$var1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function + use, click in use' => [
+            '<?php class C1 { function M1() { <sr>$var1 = 10; $f = function() use (<sr>$v<>ar1) { <sr>$var1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function (no use), click inside' => [
+            '<?php class C1 { function M1() { $var1 = 10; $f = function(<sr>$var1) { <sr>$v<>ar1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function (no use), click outside' => [
+            '<?php class C1 { function M1() { <sr>$v<>ar1 = 10; $f = function($var1) { $var1 = 5; $var2 = $var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous class method' => [
+            '<?php '.
+                'class C1 { function M1() { '.
+                '$var = 1;'.
+                '$c = new class { function IM() { <sr>$v<>ar = 1; } } '.
+                ' } }',
+            $uri,
```

Comment:
looks like we can remove the `$uri` parameter from the test as it doesn't vary?

Reply:
I would seem so. Felt bad to hardcode it in the test method as it is "data" and it is technically "provided".

Reply:
Agreed.